tasks/RTPTask.cpp

```diff
@@ -0,0 +1,352 @@
+    sender_statistics.sr_packet_count = fetchUnsignedInt(stats, "sr-packet-count");
+    sender_statistics.sent_receiver_block = fetchBoolean(stats, "sent-rb");
+    sender_statistics.sent_receiver_block_fractionlost =
+        fetchUnsignedInt(stats, "sent-rb-fractionlost");
+    sender_statistics.sent_receiver_block_packetslost =
+        fetchInt(stats, "sent-rb-packetslost");
+    sender_statistics.sent_receiver_block_exthighestseq =
+        fetchUnsignedInt(stats, "sent-rb-exthighestseq");
+    sender_statistics.sent_receiver_block_jitter =
+        fetchUnsignedInt(stats, "sent-rb-jitter");
+    sender_statistics.sent_receiver_block_lsr = ntpShortToSeconds(fetchUnsignedInt(stats, "sent-rb-lsr"));
+    sender_statistics.sent_receiver_block_dlsr = ntpShortToSeconds(fetchUnsignedInt(stats, "sent-rb-dlsr"));
+    sender_statistics.peer_receiver_reports = extractPeerReceiverReports(stats);
+
+    return sender_statistics;
+}
+
+bool RTPTask::fetchBoolean(const GstStructure* structure, const char* fieldname)
+{
+    gboolean gboolean_value;
+    if (!gst_structure_get_boolean(structure, fieldname, &gboolean_value)) {
+        return false;
+    }
+
+    return gboolean_value;
+}
+
+uint64_t RTPTask::fetch64UnsignedInt(const GstStructure* structure, const char* fieldname)
+{
+    guint64 guint64_value;
+    if (!gst_structure_get_uint64(structure, fieldname, &guint64_value)) {
+        return 0;
+    }
+
+    return guint64_value;
+}
+
+uint32_t RTPTask::fetchUnsignedInt(const GstStructure* structure, const char* fieldname)
+{
+    guint guint32_value;
+    if (!gst_structure_get_uint(structure, fieldname, &guint32_value)) {
+        return 0;
+    }
+
+    return guint32_value;
+}
+
+uint32_t RTPTask::fetchInt(const GstStructure* structure, const char* fieldname)
+{
+    gint gint_value;
+    if (!gst_structure_get_int(structure, fieldname, &gint_value)) {
+        return 0;
+    }
+
+    return gint_value;
+}
+
+std::string RTPTask::fetchString(const GstStructure* structure, const char* fieldname)
+{
+    const gchar* gstr_value = gst_structure_get_string(structure, fieldname);
+    if (gstr_value == nullptr) {
+        return std::string("");
+    }
+
+    return std::string(gstr_value);
+}
+
+int64_t RTPTask::ntpToUnixMicroseconds(uint64_t ntp_timestamp)
+{
+    // Extract seconds (upper 32 bits)
+    uint32_t ntp_seconds = ntp_timestamp >> 32;
+
+    // Extract fractional part (lower 32 bits)
+    uint32_t ntp_fraction = ntp_timestamp & 0xFFFFFFFF;
+
+    // Convert to UNIX time in seconds (1900 to 1970)
+    uint32_t unix_seconds = ntp_seconds - 2208988800;
+
+    // Convert fractional part to microseconds
+    uint64_t microseconds = (static_cast<uint64_t>(ntp_fraction) * 1000000ULL) >> 32;
+
+    // Combine seconds and microseconds
+    return (static_cast<int64_t>(unix_seconds) * 1000000ULL) + microseconds;
+}
+
+// Only gets the seconds according to documentation
```

Comment:
Secounds. Time. `base::Time`

And do *not* convert to double to do the conversion. Convert to microseconds with integer math.